submissions/Dovahkiin1991/friends-app/app.js

```diff
@@ -0,0 +1,304 @@
+const usersList = [];
+let usersFilteredList = [];
+const nameInput = document.querySelector('.filter__block input');
+
+const callAPI = async () => {
+    const data = await fetch('https://randomuser.me/api/?results=30&nat=ua');
+    handleErrors(data);
+
+    const { results: users } = await data.json();
+
+    return users;
+};
+
+const handleErrors = (response) => {
+    if (!response.ok) {
+        throw Error(response.statusText);
+    }
+    return response;
+}
+
+const getUsers = async () => {
+    try {
+        const users = await callAPI();
+        usersList.push(...users);
+        renderUsers(getSidebarOptions(), usersList);
+    } catch (e) {
+        console.log(e);
+    }
+}
+
+const mainContainer = document.getElementById('main__content');
+const usersListContainer = document.createElement('div');
+usersListContainer.classList.add('users');
+
+const renderUsers = (filtersObject, usersArray) => {
+    let users = filterUsers(filtersObject, usersArray);
+    users = users.map((user) => createUser(user)).join("");
+    usersListContainer.innerHTML = users;
+    mainContainer.innerHTML = '';
+    mainContainer.appendChild(usersListContainer);
+    window.location.hash = '';
+}
+
+const createUser = (user) => {
```

Comment:
Users aren't created in your work, they are provided by the API. Maybe `createUserCard`?